### ISSUE TITLE
WaterSwap causes protein structure to become warped

### ISSUE THREAD

**Issue by nigel-palmer-cresset** [michellab/Sire#426]
Hi, I have run WaterSwap with Sire 2023.1.1 on protein 1OIT and each time the calculation completes an iteration the structure becomes a bit more warped.
After 429 iterations the protein was looking like 
![image](https://user-images.githubusercontent.com/30503133/215735368-32ec1c6f-4a4b-4e7c-99ee-4f5ef83d021d.png)

Here are the files generated by WaterSwap, I have had to remove the \*.s3 files and some of the free_mobile_\*.pdb and bound_mobile_\*.pdb files due to GitHub file size limit.

[waterswap_23470_1.zip](https://github.com/michellab/Sire/files/10546491/waterswap_23470_1.zip)

Many Thanks

**Comment by chryswoods** [comment]
Thanks for reporting this - that definitely looks strange. I've downloaded the files and will reproduce and then search for the source of the bug :-)

**Comment by chryswoods** [comment]
I can confirm that I do see this locally. I'm debugging in the [fix_waterswap](https://github.com/OpenBioSim/sire/tree/fix_waterswap) branch in `openbiosim/sire`. I've created a simple gas-phase protein Monte Carlo script, [run_protein_py.txt](https://github.com/michellab/Sire/files/10550663/run_protein_py.txt), which runs the larger of the proteins in your test file for 500k steps. This looks stable. I suspect the issue may be something going wrong with the definition of the forcefield and its representation in sire, as I don't see any of the run-time total energy checks failing (these catch book-keeping coding errors). Likely something to do with missing protein-protein energies, or poor interactions with the grid forcefield.

I'll keep looking.

**Comment by chryswoods** [comment]
Ok - I've fixed this in [this pull request](https://github.com/OpenBioSim/sire/pull/5). It was an extremely subtle bug! More details are in the [commit message](https://github.com/OpenBioSim/sire/commit/696055eb866e7261e54b2a06cc5cf52d6dc87662). I've got a simulation running now to be sure, but comparison of component energies against 2022.1.0 and running of shorter simulations show stability.

This will go through the CI, be reviewed by @lohedges, and it will then be merged to create a new `dev` Conda package. Assuming this is working, we will trigger a release for `2023.1.2`.
